src/builtins.c

```diff
@@ -436,7 +458,10 @@ JL_CALLABLE(jl_f_apply)
             jl_value_t *argarr = jl_apply(jl_append_any_func, &args[1], nargs-1);
             assert(jl_typeis(argarr, jl_array_any_type));
             JL_GC_PUSH1(&argarr);
-            jl_value_t *result = jl_apply(f, jl_cell_data(argarr), jl_array_len(argarr));
+            size_t new_narg = jl_array_len(argarr);
+            jl_value_t **new_args = jl_cell_data(argarr);
+            jl_check_nargs_defined(new_args, new_narg);
```

Comment:
So I guess extending `Base.append_any` would be undefined behavior (just like many other non-exported functions)